### PR TITLE
Ar fixes (Added filename table)

### PR DESCRIPTION
**Pull request by xarkes**

### CODE REVIEW

File: shlr/ar/ar.c [hunk 1]
```diff
@@ -3,99 +3,69 @@
...
 	if (!b) {
 		r_sys_perror (__FUNCTION__);
 		return NULL;
 	}
 
 	/* Read magic header */
-	char *buffer = calloc (1, 1024);
-	// r = r_buf_read (b, buffer, 8);
-	r = ar_read (b, buffer, 8);
+	char *buffer = calloc (1, BUF_SIZE);
```
Comment (XVilka) [inline shlr/ar/ar.c]: Please check if error also.

File: shlr/ar/ar.c [hunk 2]
```diff
@@ -3,99 +3,69 @@
...
-		/* File content */
-		if (!strcmp (curfile, filename)) {
-			break;
+		r = ar_read_file (b, buffer, false, files, filename);
+	}
+
+	if (!filename) {
+		RListIter *iter;
+		char *name;
+		puts ("Available files:\n");
```
Comment (XVilka) [inline shlr/ar/ar.c]: Are you sure it should be `puts()`?

**Comment by radare** [comment]
Augmented reality? :D

File: shlr/ar/ar.c (hunk 2 above)
Comment (radare) [inline shlr/ar/ar.c]: Inthink thats whar zip does.. not sure if cons or cbprintf is available via bin

Comment (xarkes) [inline shlr/ar/ar.c]: r_cons is not accessible from here

**Pull request merged**